app/src/main/java/com/github/pockethub/android/util/ImageBinPoster.kt

```diff
@@ -0,0 +1,71 @@
+package com.github.pockethub.android.util
+
+import android.content.Context
+import android.net.Uri
+import io.reactivex.Single
+import okhttp3.*
+import okio.Okio
+import java.io.IOException
+
+object ImageBinPoster {
+
+    /**
+     * Post the image to ImageBin
+     *
+     * @param context A context
+     * @param uri The content URI
+     * @return Single containing the network Response
+     */
+    @JvmStatic
+    fun post(context: Context, uri: Uri): Single<Response> {
+        var bytes: ByteArray? = null
+
+        try {
+            val stream = context.contentResolver.openInputStream(uri)
+            if (stream != null) {
+                val source = Okio.source(stream)
+                bytes = Okio.buffer(source).readByteArray()
+            }
+        } catch (e: IOException) {
+            return Single.error(e)
+        }
+
+        return post(bytes)
+    }
+
+    /**
+     * Post the image to ImageBin
+     *
+     * @param bytes Bytes of the image to post
+     * @return Single containing the network Response
+     */
+    @JvmStatic
+    fun post(bytes: ByteArray?): Single<Response> {
+        val requestBody = MultipartBody.Builder()
+                .setType(MultipartBody.FORM)
+                .addFormDataPart("file", "test", RequestBody.create(MediaType.parse("image/*"), bytes!!))
+                .build()
+
+        val request = Request.Builder()
+                .url("https://imagebin.ca/upload.php")
+                .post(requestBody)
+                .build()
+
+        val client = OkHttpClient()
```

Comment:
This should not be re-created for each post. https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/OkHttpClient.java#L58-L83

Reply:
Agreed. I hadn't changed the `OkHttpClient()` instantiation process from last time but I'll make a commit to do this.